### ISSUE TITLE
how to branch someone else fork?

### ISSUE THREAD

**Issue by cctrunz** [landlab/landlab#1265]
Dear landlab team,

I would like to fork my advisor's fork on landlab. just like you can see in this picture below, where KarstModel forked speleophysics' fork of landlab:
![image](https://user-images.githubusercontent.com/30932911/106789463-e9ef7f00-6617-11eb-8dab-970b16af2c34.png)
The issue is that when I click on fork when I am on its fork, I get the message that I already have a fork of landlab (cctrunz/landlab). And I found that you can't fork 2 times with the same account (https://stackoverflow.com/questions/6675994/is-it-possible-to-fork-a-fork-in-github)
Is there anyway to create a separate branch that's disconnected from my cctrunz/landlab branch and based of speleophysics fork?
What would you recommand I do?

For now, I tried to branch a branch of the speleophysics fork of landlab to work on it. This created a new branch on my personal fork of landlab, and now I'm getting a warning message on my github desktop. I'm unsure what to do now.
![GitHub Desktop 03 02 2021 11_19_06](https://user-images.githubusercontent.com/30932911/106786628-608a7d80-6614-11eb-9a69-0b877b341594.png)

Thanks a lot for your help. 
Celia

**Comment by gregtucker** [comment]
Hi Celia, I think you have a couple of options:
* Work on Matt's fork (he would have to invite you to collaborate on the repo; I don't remember the details of how to do this but it's possible), via a branch(es) of your own -- i.e., his repo would kind of be the lab's repo
* Delete your current fork and make a new one (if there's stuff in the old one you want to keep, you could do a PR to merge it back into speleophysics' fork first, and then it will be in your new fork)

Fork-of-a-fork should probably be avoided in general, since it gets increasingly complicated to merge back in with main version.

That said, I know just enough git lore to be dangerous; you'd probably get better advice from @mdpiper

**Comment by mdpiper** [comment]
Hi, Celia! What you want are separate *remotes* of the landlab repository. I'll make an example & post it here. It's really neat (I swear).

**Comment by mdpiper** [comment]
*Remotes* (shorthand for remote repositories) are just versions of the same repository in different locations.

I've forked the landlab repository and cloned it to my computer. This means there are now three remotes I can address:

* *local* -- the repository on my machine
* *origin* -- my fork on GitHub
* *upstream* -- the landlab repo on GitHub

We can use the `git remote` subcommand to work with remotes. Here's what it looks like on my computer:
```
$ git remote -v
origin	git@github.com:mdpiper/landlab.git (fetch)
origin	git@github.com:mdpiper/landlab.git (push)
```
Note that *local* is assumed, and *upstream* hasn't been set. I can add it with:
```
$ git remote add upstream https://github.com/landlab/landlab
```
Now my remotes are:
```
$ git remote -v
origin	git@github.com:mdpiper/landlab.git (fetch)
origin	git@github.com:mdpiper/landlab.git (push)
upstream	https://github.com/landlab/landlab (fetch)
upstream	https://github.com/landlab/landlab (push)
```

Now let me add your advisor's landlab fork to my list of remotes. I'll call the remote *speleophysics*, because thet's an awesome name:
```
git remote add speleophysics https://github.com/speleophysics/landlab
```
Check my current remotes:
```
$ git remote -v
origin	git@github.com:mdpiper/landlab.git (fetch)
origin	git@github.com:mdpiper/landlab.git (push)
speleophysics	https://github.com/speleophysics/landlab (fetch)
speleophysics	https://github.com/speleophysics/landlab (push)
upstream	https://github.com/landlab/landlab (fetch)
upstream	https://github.com/landlab/landlab (push)
```

If I'd like to get a branch from *speleophysics*, I can `git fetch` from their remote:
```
$ git fetch speleophysics
...(lots of output)...
```
and then list all the available branches:
```
$ git branch -r
...(lots of output)...
```
I'll check the one called `vegetation_ca`:
```
$ git checkout speleophysics/vegetation_ca
```

Whew, I know this is long, but the idea is you need only fork, but it can reference multiple remotes.

A nice reference for working with remotes is provided in the [git documentation](https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes).

**Comment by cctrunz** [comment]
Thank you both sooo much for your quick and enthusiastic answer!

Marc, you solution works perfectly! this is exactly what I needed. 

What I did is that I removed the branch I previously create and that was giving me a warning message. 
Then I followed your steps and checkout the speleophysics branch I was interested in.  Because I want to work on it, I added this step that was suggested in the bash, and I create a new branch based out of my advisor's branch with:
`$ git switch -c my_new_branch_based_of_speleophyics_flow_branch`

Now, I can move back and forth from my own landlab fork to the branch that I made based out of speleophysics, push all my changes, and it updates my landlab folder on my computer. This is magic! Thank you so much!
